src/transactions/TxEnvelopeTests.cpp

```diff
@@ -199,16 +199,51 @@ TEST_CASE("txenvelope", "[tx][envelope]")
 
         SECTION("success two signatures")
         {
-            auto tx = a1.tx({payment(root, 1000)});
+            // updating thresholds requires high
+            auto tx = a1.tx(
+                {setOptions(nullptr, nullptr, nullptr, &th, &sk1, nullptr)});
+
             tx->getEnvelope().signatures.clear();
             tx->addSignature(s1);
             tx->addSignature(s2);
 
             for_all_versions(app, [&] {
                 applyCheck(tx, app);
                 REQUIRE(tx->getResultCode() == txSUCCESS);
-                REQUIRE(PaymentOpFrame::getInnerCode(getFirstResult(*tx)) ==
-                        PAYMENT_SUCCESS);
+                REQUIRE(SetOptionsOpFrame::getInnerCode(getFirstResult(*tx)) ==
+                        SET_OPTIONS_SUCCESS);
+            });
+        }
+
+        SECTION("without master key")
+        {
+            // payment requires medium
+            auto tx = a1.tx({payment(root, 1000)});
+            // only sign with s2 (med)
+            tx->getEnvelope().signatures.clear();
+            tx->addSignature(s2);
+
+            ThresholdSetter th2;
+            th2.masterWeight = make_optional<int>(0);
+            auto setOptionsTx = a1.tx({setOptions(nullptr, nullptr, nullptr,
+                                                  &th2, nullptr, nullptr)});
+            setOptionsTx->addSignature(s2);
```

Comment:
this transaction is never executed - it is only created as transaction object, you need to call applyCheck to execute it

Reply:
ah yes - I missed that the setup had the `masterweight` with a weight of `high`.
I reworked the test a bit, should be closer to what I originally wanted.